ckan/logic/validators.py

```diff
@@ -562,6 +562,59 @@ def ignore_not_sysadmin(key: FlattenKey, data: FlattenDataDict,
     data.pop(key)
 
 
+def limit_sysadmin_update(key: FlattenKey, data: FlattenDataDict,
+                          errors: FlattenErrorDict, context: Context) -> Any:
+    """
+    Should not be able to modify your own sysadmin privs, or the system user's
+    """
+    value = data.get(key)
+
+    if value is None:
+        # sysadmin key not in data, return here
+        data.pop(key, None)
```

Comment:
I think validators should not remove the key from the validated data unless they are of type "ignore if", where you actually want to remove it from further validation. In this case, other validators might need the key to still be there, even if empty. Same for the other places where the key is popped

Reply:
Ah okay, sounds good. Will update to not have any pops in this validator